test/Microsoft.AspNetCore.Server.Kestrel.Performance/ResponseHeaders.cs

```diff
@@ -0,0 +1,215 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using Microsoft.AspNetCore.Hosting.Internal;
+using Microsoft.AspNetCore.Hosting.Server;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Server.Kestrel.Internal;
+using Microsoft.AspNetCore.Server.Kestrel.Internal.Http;
+using Microsoft.AspNetCore.Server.Kestrel.Internal.Infrastructure;
+using Microsoft.AspNetCore.Testing;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.ObjectPool;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Performance
+{
+    [Config(typeof(CoreConfig))]
+    public class ResponseHeaders
+    {
+        private const int InnerLoopCount = 512;
+
+        private static FrameResponseHeaders HeadersDirect;
+        private static DateHeaderValueManager DateHeaderValueManager = new DateHeaderValueManager();
+
+        private MemoryPool MemoryPool;
+        private SocketInput SocketInput;
+        private HostingApplication Application;
+        private HostingApplication.Context Context;
+
+        [Params("ContentLengthNumeric", "ContentLengthString", "Plaintext", "Primary", "Common", "Unknown")]
+        public string Type { get; set; }
+
+        private RequestDelegate GetRequestDelegate(string type)
+        {
+            switch (type)
+            {
+                case "Plaintext":
+                    return Plaintext;
+                case "ContentLengthString":
+                    return ContentLengthString;
+                case "ContentLengthNumeric":
+                    return ContentLengthNumeric;
+                case "Primary":
+                    return Primary;
+                case "Common":
+                    return Common;
+                case "Unknown":
+                    return Unknown;
+            }
+
+            return Plaintext;
+        }
+
+        [Benchmark(OperationsPerInvoke = InnerLoopCount)]
+        public async Task BenchmarkAsync()
+        {
+            for (int i = 0; i < InnerLoopCount; i++)
+            {
+                await Application.ProcessRequestAsync(Context);
```

Comment:
Why benchmark via app instead of just instantiating `FrameResponseHeaders` and setting directly?